modules/Emails/EmailsController.php

```diff
@@ -627,4 +651,49 @@ protected function setAfterImport($importedEmailId, $request)
 
         return $emails;
     }
+
+    /**
+     * @param User $requestedUser
+     * @param InboundEmail $requestedInboundEmail
+     * @param Email $requestedEmail
+     * @return bool false if user doesn't have access
+     */
+    protected function userIsAllowedToSendEmail($requestedUser,  $requestedInboundEmail, $requestedEmail)
+    {
+        $hasAccess = false;
+        $hasAccessToInboundEmailAccount = false;
+
+        // Check that user is allowed to use inbound email account
+        $usersInboundEmailAccounts = $requestedInboundEmail->retrieveByGroupId($requestedUser->id);
```

Comment:
Note about the called function: Field group_id in the table inbound_email is not indexed. New issue?